atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBackupRunner.java

```diff
@@ -43,8 +44,10 @@
 
     private final CassandraKeyValueService cassandraKeyValueService;
 
-    public CassandraTimestampBackupRunner(CassandraKeyValueService cassandraKeyValueService) {
-        this.cassandraKeyValueService = cassandraKeyValueService;
+    public CassandraTimestampBackupRunner(KeyValueService keyValueService) {
+        Preconditions.checkArgument(keyValueService instanceof CassandraKeyValueService,
```

Comment:
Hmm, you're right - if you're instantiating this class it's reasonable to assume that you're solidly in the cassandra world, whereas with invalidators (a common interface across KVSs) it could more plausibly go wrong.

I haven't tested the 'bad' case, will add a test for that.